pkg/controller/account/account_controller.go

```diff
@@ -165,25 +163,50 @@ func (r *ReconcileAccount) Reconcile(request reconcile.Request) (reconcile.Resul
 		}
 	}
 
-	// We expect this secret to exist in the same namespace Account CR's are created
-	awsSetupClient, err := awsclient.GetAWSClient(r.Client, awsclient.NewAwsClientInput{
-		SecretName: utils.AwsSecretName,
-		NameSpace:  awsv1alpha1.AccountCrNamespace,
-		AwsRegion:  "us-east-1",
-	})
+	// If account CR is generated by accountpool_controller, there will be no ClaimLink available. If
+	// this field is empty set region to us-east-1 as AWS China only supports CCS/BYOC.
+	var awsRegion string
+	accountClaim := &awsv1alpha1.AccountClaim{}
+	if currentAcctInstance.Spec.ClaimLink == "" {
+		// Set region to us-east-1 if ClaimLink is not empty.
+		awsRegion = awsv1alpha1.AwsPlatformConfigGlobal.ClientInput.AwsRegion
+	}
+
+	// Return associated accountClaim.
+	accountClaim, err = r.ReturnAccountClaim(currentAcctInstance)
+	if err != nil {
+		reqLogger.Error(err, "Failed to return account from accountClaim")
+		return reconcile.Result{}, err
+	}
+
+	// Get region AWS Region to determine AWS Global or China platform.
+	awsRegion = accountClaim.Spec.Aws.Regions[0].Name
+
+	// Based on the awsRegion bool, import either the Global or China configuration.
+	AwsPlatformConfig := awsv1alpha1.AwsPlatformConfig{}
+
+	if !checkAwsProviderIsGlobal(awsRegion) {
+		AwsPlatformConfig = awsv1alpha1.AwsPlatformConfigChina
+	} else {
+		AwsPlatformConfig = awsv1alpha1.AwsPlatformConfigGlobal
+	}
+	reqLogger.Info(fmt.Sprintf("AWS Platform configured region: %s", AwsPlatformConfig.ClientInput.AwsRegion))
```

Comment:
@dofinn Currently there are 1000's of Account CR's that don't have claims or have a `.spec.ClaimLink` that points to non existing claims. When the accountpool_controller creates Account CRs satisfying the pool they don't have claims, they are awaiting to be claimed. Merging this as is would have caused the account_controller to continually error since it would requeue that Account CR with no ClaimLink and just error again forever.